misk-feature/src/main/kotlin/misk/feature/RealAppScopedDynamicConfigResolver.kt

```diff
@@ -0,0 +1,31 @@
+package misk.feature
+
+import misk.logging.getLogger
+
+internal class RealAppScopedDynamicConfigResolver<T : ValidatableConfig<T>> constructor(
+  private val appName: String,
+  private val configName: String,
+  private val dynamicConfig: DynamicConfig,
```

Comment:
i could see this being useful for `FeatureFlags` as well. For example you have a single feature flag for all your topic configs and the flag is keyed on topic. that's what we did for jobqueues.

Reply:
Hm, this makes me think that `FeatureFlags` would be preferable to `DynamicConfig` as the basis of this... Forcing access without key inputs might force app owners to maintain a single flag mega-config object (e.g., nested config) _or_ to add numerous feature flags (instead of single variations using a single key). 

I could see it being nice to have a few separate features for configuring different app components, but it would be way better to have N event consumer's `ConsumerConfig`s provided by a single flag given the topic as an input than to compose all of them into one global "all event consumers" config feature.

Reply:
just expose an api that allows both key and non-key lookups. under the hood you'll need to use `FeatureFlags` probably.

Reply:
For this use case, you should probably inject `LDClientInterface` directly. You don't need the `FeatureFlags` abstraction